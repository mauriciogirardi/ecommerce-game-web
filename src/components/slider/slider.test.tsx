import '../../../tests/match-media-mock'

import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import { Slider } from '.'

describe('<Slider />', () => {
  it('should be able to render the children as slider items', () => {
    const { container } = render(
      <Slider
        settings={{
          slidesToShow: 1,
          dots: false,
          slidesToScroll: 1,
          infinite: false
        }}
      >
        <div>item 1</div>
        <div>item 2</div>
      </Slider>
    )

    expect(screen.getByText('item 1').parentElement?.parentElement).toHaveClass(
      'slick-slide'
    )
    expect(screen.getByText('item 2').parentElement?.parentElement).toHaveClass(
      'slick-slide'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
