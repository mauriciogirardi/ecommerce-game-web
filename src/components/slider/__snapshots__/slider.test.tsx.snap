// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Slider /> > should be able to render the children as slider items 1`] = `
<div
  class="slick-slider slick-initialized"
  dir="ltr"
>
  <button
    class="slick-arrow slick-prev slick-disabled"
    data-role="none"
    style="display: block;"
    type="button"
  >
     
    Previous
  </button>
  <div
    class="slick-list"
  >
    <div
      class="slick-track"
      style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
    >
      <div
        aria-hidden="false"
        class="slick-slide slick-active slick-current"
        data-index="0"
        style="outline-color: none; outline-style: none; outline-width: initial; width: 0px;"
        tabindex="-1"
      >
        <div>
          <div
            style="width: 100%; display: inline-block;"
            tabindex="-1"
          >
            item 1
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="slick-slide"
        data-index="1"
        style="outline-color: none; outline-style: none; outline-width: initial; width: 0px;"
        tabindex="-1"
      >
        <div>
          <div
            style="width: 100%; display: inline-block;"
            tabindex="-1"
          >
            item 2
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="slick-arrow slick-next"
    data-role="none"
    style="display: block;"
    type="button"
  >
     
    Next
  </button>
</div>
`;
